### PR TITLE
node: recent crypto updates

### PR DESCRIPTION
**Pull request by panva**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/releases/tag/v15.7.0, https://github.com/nodejs/node/releases/tag/v15.9.0, https://github.com/nodejs/node/releases/tag/v15.12.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by panva** [comment]
`mongoose` tests failure is unrelated to this PR.

File: types/node/crypto.d.ts [hunk 1]
```diff
@@ -1174,6 +1174,12 @@ declare module 'crypto' {
         data: NodeJS.ArrayBufferView,
         key: KeyLike | SignKeyObjectInput | SignPrivateKeyInput,
     ): Buffer;
+    function sign(
+        algorithm: string | null | undefined,
+        data: NodeJS.ArrayBufferView,
+        key: KeyLike | SignKeyObjectInput | SignPrivateKeyInput,
+        callback: (err: Error | null, signature: Buffer) => any,
```
Comment (SimonSchick) [inline types/node/crypto.d.ts:1181]: Callback return type should be `void`.

File: types/node/crypto.d.ts [hunk 2]
```diff
@@ -1189,6 +1195,13 @@ declare module 'crypto' {
         key: KeyLike | VerifyKeyObjectInput | VerifyPublicKeyInput,
         signature: NodeJS.ArrayBufferView,
     ): boolean;
+    function verify(
+        algorithm: string | null | undefined,
+        data: NodeJS.ArrayBufferView,
+        key: KeyLike | VerifyKeyObjectInput | VerifyPublicKeyInput,
+        signature: NodeJS.ArrayBufferView,
+        callback: (err: Error | null, result: boolean) => any,
```
Comment (SimonSchick) [inline types/node/crypto.d.ts:1203]: See above.

**Review by SimonSchick — CHANGES_REQUESTED** [review]
This generally looks good but I'm looking to get these merged in order to avoid version confusion, I filed a PR to get up 15.2 but we are still a good bit away from 15.12.

I think we can reasonable merge this in a week, with an added note it's only available as of 15.12

**Comment by SimonSchick** [comment]
Also please consider merging https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52653 into this as it is easier to keep track of the versioning this way.

**Comment by SimonSchick** [comment]
@panva I'm making good progress getting things up to date, see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52732

I'm trying to batch version changes together, do you have any objections of me integrating your changes into my branch?

**Comment by panva** [comment]
No objections

**Comment by SimonSchick** [comment]
Let me know if you have any other fragments you have been working on, I'd be glad to integrate them as well.

**Pull request closed without merging**